### PR TITLE
Wrap SCNetworkReachabilityFlags and fix Catalyst warning

### PR DESCRIPTION
**Pull request by doudounan**
1 Wrap SCNetworkReachabilityFlags in GDTCORPlatform.
2 Fix Catalyst warning in GDTCORPlatform.

### CODE REVIEW

**Comment by chliangGoogle** [comment]
Can you also add watchos along with other platforms in https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/datatransport.yml? This helps prevent errors again like presubmit.

**Comment by doudounan** [comment]
> Can you also add watchos in https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/datatransport.yml? This helps prevent errors again like presubmit.

Yes, we definitely should.

**Comment by mikehaney24** [comment]
> #4955

Standard pod lib lint doesn't do anything, so making this change doesn't do anything. watchOS also doesn't support unit tests. The reason why it works in storage is because they use a different build and test system. Doudou or myself would have to replicate that infra here in GDT.

pod gen can't even generate a watchOS project, so I'm not sure anything is building.

```
 -> GoogleDataTransport
[!] Skipping test spec `GoogleDataTransport/Tests-Unit` on platform `watchos` since it is not supported.
 -> GoogleDataTransport (4.0.0)
    - NOTE  | xcodebuild:  note: Using new build system
```

File: GoogleDataTransport/GDTCORLibrary/Public/GDTCORPlatform.h [hunk 1]
```diff
@@ -68,6 +75,12 @@ typedef NS_ENUM(NSInteger, GDTCORNetworkMobileSubtype) {
 BOOL GDTCORReachabilityFlagsContainWWAN(SCNetworkReachabilityFlags flags);
 #endif
 
+/** Generates an enum message GDTCORNetworkType representing Network connection type.
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORLibrary/Public/GDTCORPlatform.h]: network

**Comment by chliangGoogle** [comment]
I see. Isn't storage just using the new github action? I thought as long as we copied the yml file, we've used the same build and test system.

**Comment by mikehaney24** [comment]
> I see. Isn't storage just using the new github action? I thought as long as we copied the yml file, we've used the same build and test system.

They're still using GHA for CI, but it calls into `scripts/build.sh`, whereas GDT's workflow calls `scripts/pod_lib_lint.sh`

For GDT, we'd need to probably add something back to build.sh for our watchOS build that builds and runs a manually created watchOS test project

**Comment by chliangGoogle** [comment]
I was planning just duplicate the yml in messaging, looks like there are more work needs to be done here. Thanks for the explanation! I will look into more for messaging as well.

**Pull request merged**